### ISSUE TITLE
Need help on Systemd start-on-boot file

### ISSUE THREAD

**Issue by Manic137** [vincss/mcsleepingserverstarter#74]
Hey there, I would like to write a start-on-boot Systemd Script, but I have no idea what to put in the "ExecStop" field, any thoughts?

This is what I used before, but I did change the "ExecStart" already to what I think will work for starting the server, however, I'm somewhat lost as to what to write for "ExecStop":

```
[Unit]
Description=Starts PaperMC Server
After=network-online.target
Wants=network-online.target

[Service]
Type=simple
User=pmcserver
WorkingDirectory=/home/pmcserver/serverfiles
ExecStart=/home/pmcserver/serverfiles/mcsleepingserverstarter-linux
ExecStop=/home/pmcserver/serverfiles/
Restart=no
RemainAfterExit=yes

[Install]
WantedBy=multi-user.target
```

**Comment by Manic137** [comment]
Update #1: This is what I've tested, I have made a code based on some basic experience of writing scripts, but it did not work as intended. The server started successfully, both the program and server, but the stop command did not work. Also, I was having issues with the server sleeping after no players are in game, this might be a separate issue, more testing is required. I will paste the Minecraft.service script below;

Script:
```
[Unit]
Description=Starts PaperMC Server
After=network-online.target
Wants=network-online.target

[Service]
Type=simple
User=pmcserver
WorkingDirectory=/home/pmcserver/serverfiles
ExecStart=/home/pmcserver/serverfiles/mcsleepingserverstarter-linux
ExecStop=/home/pmcserver/serverfiles/mcsleepingserverstarter-linux stop
Restart=no
RemainAfterExit=yes

[Install]
WantedBy=multi-user.target
```

**Comment by Manic137** [comment]
Still waiting on help :(

**Comment by vincss** [comment]
Hi !
I missed your issue.
Maybe you can try to start the exec when the system starts.
I use crontab for that.

Logging with the pmcserver user; then 
contab -e
add a line 
@reboot  /home/pmcserver/serverfiles/start.sh

Create the script with something like : 
nano /home/pmcserver/serverfiles/start.sh
`#!/bin/bash
folder=$(dirname "$0")
cd  $folder

msg="$(date --iso-8601=seconds) ... Starting screen and server ... $(whoami) -> $(pwd) "
echo $msg
echo $msg >> start.log


./mcsleepingserverstarter-linux &`


Make the script executable with 
chmod +x start.sh

And reboot


For the issue with no player for emptyserverstopper maybe set the parameter : 
ShutdownAtStart: true

**Comment by Manic137** [comment]
> Hi ! I missed your issue.
> 
> For the stop, you can check in the logs that the server received the stop command correctly from the system: `2022-01-21 19:19:47 info: SIGINT signal received. 2022-01-21 19:19:47 info: Cleaning up the place. 2022-01-21 19:19:47 info: [McJava] Closing 2022-01-21 19:19:47 info: [McJava] Closed 2022-01-21 19:19:47 info: [BedRock] Closing 2022-01-21 19:19:47 info: [BedRock] Socket Closed 2022-01-21 19:19:47 info: ... To be continued ...`
> 
> Otherwise, maybe you can try to start the exec when the system starts. I use crontab for that.
> 
> Logging with the pmcserver user; then contab -e add a line @reboot /home/pmcserver/serverfiles/start.sh
> 
> Create the script with something like : nano /home/pmcserver/serverfiles/start.sh `#!/bin/bash folder=$(dirname "$0") cd $folder
> 
> msg="$(date --iso-8601=seconds) ... Starting screen and server ... $(whoami) -> $(pwd) " echo $msg echo $msg >> start.log
> 
> ./mcsleepingserverstarter-linux &`
> 
> Make the script executable with chmod +x start.sh
> 
> And reboot
> 
> For the issue with no player for emptyserverstopper maybe set the parameter : ShutdownAtStart: true

Not sure if I understand the solution. I'll try out the new boot script you suggested with crontab, but, how do I stop the mcsleepingserverstarter after?

**Comment by vincss** [comment]
It would stop by it self on system close with the SIGINT signal I think.
But I don't know how it should stop properly when MC is running.

**Comment by Manic137** [comment]
> It would stop by it self on system close with the SIGINT signal I think. But I don't know how it should stop properly when MC is running.

Is the SIGINT signal a shut down of the PC? Moreover wondering how to stop the service with a command, but it's okay, I'd imagine I don't need to since the server can shut itself down. Still tho, I'm curious if it is possible to set up a way to turn off the wake up service by command

**Comment by vincss** [comment]
Yes, you can do : 
killall -s SIGINT mcsleepingserverstarter-linux

**Comment by Manic137** [comment]
> Yes, you can do : killall -s SIGINT mcsleepingserverstarter-linux

Thank you!
